File: Js/Other Algo/Two_pointer_fromleftside.js
function countUniqueValues(arr){
  var i = 0;
  for(var j=1; j<arr.length; j++){
    if(arr[i] !== arr[j]){
      i++;
      arr[i] = arr[j]
    }
    console.log(i,j)
  }
  //BEcause i is the index position in array so we increment by one to get the result
  return i+1
}

//console.log(countUniqueValues([]))
console.log(countUniqueValues([1,1,2,3,3,3,4]));

File: Python/Other algo/two_pointer_fromleftside.py
def checkUniqueValues(arr):
    i = 0
    for j in range(1,len(arr)):
        if arr[i] != arr[j]:
            i += 1
            arr[i] = arr[j]
        
    return i + 1

print(checkUniqueValues([1,1,2,3,3,3,4]))